shared/react-native/ios/KeybaseShare/FilesViewController.m

```diff
@@ -0,0 +1,222 @@
+  }
+}
+
+bool filterWritableEntries(NSDictionary* entry) {
+  return [entry objectForKey:@"writable"];
+}
+
+- (void)parseFiles:(NSString*)jsonFiles {
+  NSError* error = nil;
+  NSData* data = [jsonFiles dataUsingEncoding:NSUTF8StringEncoding];
+  NSArray* items = [NSJSONSerialization JSONObjectWithData:data options: NSJSONReadingMutableContainers error: &error];
+  NSArray* sortedAndFilteredItems;
+  if (!items) {
+    if (error) {
+      NSLog(@"parseFiles: error parsing JSON: %@", error);
+    }
+    // At least show an empty folder.
+    sortedAndFilteredItems = [[NSArray alloc] init];
+  } else {
+    // Sort items: directories first, then alphabetically.
+    sortedAndFilteredItems = [items sortedArrayUsingFunction:sortEntries context:NULL];
+    
+    // For paths deeper than 2, filter out folders that aren't writable.
+    unsigned long pathLength = [self pathLength];
+    if (pathLength > 2) {
+      sortedAndFilteredItems = [sortedAndFilteredItems filteredArrayUsingPredicate:[NSPredicate predicateWithBlock:^BOOL(id entry, NSDictionary* bindings) {
+        return [entry objectForKey:@"writable"];
+      }]];
+    }
+  }
+  if ([self.path isEqualToString:@"/"]) {
+    // If we're at the root, don't show a back navigation item.
+    [self setDirectoryEntries:sortedAndFilteredItems];
+  } else {
+    // Otherwise, prepend a back navigation item.
+    NSMutableArray* itemsWithBack = [[NSMutableArray alloc] init];
+    [itemsWithBack addObject:[[NSDictionary alloc] initWithObjectsAndKeys:
+                              UpOneLevel, @"name",
+                              [NSNumber numberWithInt:1], @"direntType",
+                              nil]];
+    [itemsWithBack addObjectsFromArray:sortedAndFilteredItems];
+    [self setDirectoryEntries:itemsWithBack];
+  }
+}
+
+- (void)didReceiveMemoryWarning {
+  KeybaseExtensionForceGC();
+  [super didReceiveMemoryWarning];
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+  return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+  return [self.directoryEntries count];
+}
+
+- (NSDictionary*)getItemAtIndex:(NSIndexPath*)indexPath {
+  NSInteger index = [indexPath item];
+  return self.directoryEntries[index];
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+  UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"ConvCell"];
+  if (NULL == cell) {
+    cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"ConvCell"];
+  }
+  NSDictionary* item = [self getItemAtIndex:indexPath];
+  [[cell textLabel] setText:item[@"name"]];
+  if ([[item objectForKey:@"direntType"] intValue] == 1) {
+    // Style folders differently from files. Use Keybase blue.
+    [[cell textLabel] setTextColor:[UIColor colorWithRed:76.0/255.0 green:142.0/255.0 blue:1 alpha:1]];
```

Comment:
Not suggesting a change as it's ultimately design's call, but I think in the main app it's just font-weight difference. Color is the same.